src/models/Room.ts

```diff
@@ -0,0 +1,19 @@
+import { Schema, model, Types } from 'mongoose';
+
+interface Room {
+    name: string;
+    sections: Types.Array<string>;
+    schedule: Types.Array<Record<string, string[]>>;
```

Comment:
What is a mongoose `Record` type and why use it here? Would it be better to create a Schedule interface that represents how a schedule objects will actually look?

Reply:
yeah, that will be easier, I will work on that